spec/intl.html

```diff
@@ -189,10 +189,13 @@ <h1>
     <h1><a href="https://tc39.es/ecma402/#sec-datetimeformat-abstracts">Abstract Operations For DateTimeFormat Objects</a></h1>
 
     <emu-clause id="sec-temporal-initializedatetimeformat" aoid="InitializeDateTimeFormat">
-      <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
+      <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ [ , <ins>_toLocaleStringTimeZone_</ins> ] )</h1>
 
       <p>
-        The abstract operation InitializeDateTimeFormat accepts the arguments _dateTimeFormat_ (which must be an object), _locales_, and _options_. It initializes _dateTimeFormat_ as a DateTimeFormat object. This abstract operation functions as follows:
+        The abstract operation InitializeDateTimeFormat accepts the arguments _dateTimeFormat_ (which must be an object), _locales_, and _options_.
+        It initializes _dateTimeFormat_ as a DateTimeFormat object.
+        <ins>If an additional _toLocaleStringTimeZone_ argument is provided, the time zone will be overridden and some adjustments will be made to the defaults in order to implement the behaviour of `Temporal.ZonedDateTime.prototype.toLocaleString`.</ins>
```

Comment:
```suggestion
        <ins>If an additional _toLocaleStringTimeZone_ argument is provided (which, if present, must be a canonical time zone name string), the time zone will be overridden and some adjustments will be made to the defaults in order to implement the behaviour of `Temporal.ZonedDateTime.prototype.toLocaleString`.</ins>
```

Reply:
We have this a bit below here:

> If toLocaleStringTimeZone is present, then
> - If IsAvailableTimeZoneName(toLocaleStringTimeZone) is false, throw a RangeError exception.

Is your suggestion that that should be an assertion instead, and ZonedDateTime.p.toLocaleString should ensure IsAvailableTimeZoneName before calling InitializeDateTimeFormat? That's certainly possible, though a bit asymmetric with validating the value of the `timeZone` option in this operation.

Reply:
I made this comment because `CanonicalizeTimeZoneName` is not being invoked on the new argument.

Reply:
I see your point, I've moved the IsAvailableTimeZoneName check and CanonicalizeTimeZoneName to the caller.

That also made me realize that we didn't handle offset time zones, which can't be passed to this function. So I've also handled those in the caller.